editoast/src/views/timetable/train_schedule.rs

```diff
@@ -713,6 +720,124 @@ async fn project_path(
     Ok(Json(project_path_result))
 }
 
+#[utoipa::path(
+    post, path = "",
+    tag = "train_schedule",
+    request_body = ProjectPathOpForm,
+    responses(
+        (status = 200, description = "Project train schedules on a list of operational points.", body = HashMap<i64, Vec<SpaceTimeCurve>>),
+    ),
+)]
+async fn project_path_op(
+    State(AppState {
+        db_pool,
+        valkey: valkey_client,
+        core_client,
+        ..
+    }): State<AppState>,
+    Extension(auth): AuthenticationExt,
+    Json(ProjectPathOpForm {
+        infra_id,
+        train_ids,
+        electrical_profile_set_id,
+        operational_points_ids,
+        operational_points_distances,
+    }): Json<ProjectPathOpForm>,
+) -> Result<Json<HashMap<i64, Arc<SpaceTimeCurves>>>> {
+    let infra = &Infra::retrieve_real_or_fail(db_pool.get().await?, infra_id, || {
+        TrainScheduleError::InfraNotFound { infra_id }
+    })
+    .await?;
+
+    let authorized = auth
+        .check_roles([authz::Role::OperationalStudies, authz::Role::Stdcm].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+
+    auth.check_authorization(async |authorizer| {
+        authorizer
+            .authorize_infra_read(&authz::Infra(infra_id))
+            .await
+    })
+    .await?;
+
+    let conn = &mut db_pool.get().await?;
+
+    let trains_schedules: Vec<models::TrainSchedule> =
+        models::TrainSchedule::retrieve_batch_or_fail(conn, train_ids, |missing| {
+            TrainScheduleError::BatchTrainScheduleNotFound {
+                number: missing.len(),
+            }
+        })
+        .await?;
+
+    let path_item_locations: Vec<&PathItemLocation> = trains_schedules
+        .iter()
+        .flat_map(|ts| ts.path.iter().map(|p| &p.location))
+        .collect();
+
+    let path_item_cache = PathItemCache::load(conn, infra_id, &path_item_locations).await?;
+
+    // 1. Build the projection_op_id_to_positions
+    let mut init_hashmap: HashMap<_, _> = HashMap::new();
+    init_hashmap.insert(operational_points_ids[0].clone(), 0);
+
+    let (projection_op_id_to_positions, _) = operational_points_ids
+        .into_iter()
+        .skip(1) // Skip the first one, as it is position 0 for first operational point id
+        .zip(operational_points_distances.into_iter())
```

Comment:
`zip` takes something that implements `IntoInterator` so it will call `.into_iter()` for you.


```suggestion
        .zip(operational_points_distances)
```